main.go

```diff
@@ -104,14 +111,23 @@ func main() {
 		}
 	}
 
+	pluginHandle, err := opts.Plugins.Handle()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if pluginHandle != nil {
+		defer pluginHandle.Close()
```

Comment:
can you just always `defer pluginHandle.Close()` and handle the `nil` case in the custom type.